drivers/SmartThings/zigbee-switch/src/wallhero/init.lua

```diff
@@ -0,0 +1,125 @@
+local stDevice = require "st.device"
+local zcl_clusters = require "st.zigbee.zcl.clusters"
+
+local Scenes = zcl_clusters.Scenes
+
+local FINGERPRINTS = {
+  { mfr = "WALL HERO", model = "M4T2828A-ZB", switches = 4, buttons = 4 }
+}
+
+local function can_handle_wallhero_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local function get_children_info(device)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_model() == fingerprint.model then
+      return fingerprint.switches, fingerprint.buttons
+    end
+  end
+end
+
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%02X", ep_id))
+end
+
+local function create_child_devices(driver, device)
+  local switch_amount, button_amount = get_children_info(device)
+  local base_name = device.label:sub(1, device.label:find(" "))
+  -- Create Switch 2-4
+  for i = 2, switch_amount, 1 do
+    if find_child(device, i) == nil then
+      local metadata = {
+        type = "EDGE_CHILD",
+        parent_assigned_child_key = string.format("%02X", i),
+        label = base_name .. i .. " (普通开关)",
+        profile = "basic-switch",
+        parent_device_id = device.id,
+        vendor_provided_label = base_name .. i .. " (普通开关)",
+      }
+      driver:try_create_device(metadata)
+    end
+  end
+  -- Create Button if necessary
+  for i = switch_amount+1, switch_amount+button_amount, 1 do
+    if find_child(device, i) == nil then
+      local metadata = {
+        type = "EDGE_CHILD",
+        parent_assigned_child_key = string.format("%02X", i),
+        label = base_name .. i .. " (场景开关)",
+        profile = "wallhero-scene-switch",
+        parent_device_id = device.id,
+        vendor_provided_label = base_name .. i .. " (场景开关)",
+      }
+      driver:try_create_device(metadata)
+    end
+  end
+  device:refresh()
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= stDevice.NETWORK_TYPE_CHILD then
+    create_child_devices(driver, device)
+  end
+  device:emit_event(capabilities.button.numberOfButtons({ value = 4 }, { visibility = { displayed = false } }))
+  device:emit_event(capabilities.button.supportedButtonValues({ "pushed" }, {visibility = {displayed = false } }))
+end
+
+local function device_init(driver, device, event)
+  device:set_find_child(find_child)
+end
+
+local function scenes_cluster_handler(driver, device, zb_rx)
+  log.info("Enter scenes_cluster_handler")
+  local additional_fields = {
+    state_change = true
+  }
+  local ep = zb_rx.address_header.src_endpoint.value
+  local child_device = device:get_child_by_parent_assigned_key(string.format("%02X", ep))
+  if child_device ~= nil then
+    local event = capabilities.button.button.pushed(additional_fields)
+    child_device:emit_event(event)
```

Comment:
you can just call `emit_event_for_endpoint` which will call the `find_child` function you set in `init`